bika/lims/browser/sample/printform.py

```diff
@@ -59,21 +63,43 @@ class SamplesPrint(BrowserView):
     _avoid_filter_by_date = False
 
     def __call__(self):
-        if self.context.portal_type == 'SamplesFolder':
+        # When printing sample sheets this method is called at least
+        # twice.
+        # The first time it gets the uids of the selected samples
+        # in the samples form and filters them by workflow status.
+        # The result is the list of samples to be printed.
+        # Then it builds the URL that will print the sample
+        # sheets (including the previous list as the items parameter
+        # of the request) and redirects to it. This results in this
+        # method being called again. The first lines are skipped because
+        # the form now is empty. In this second execution the
+        # actual rendering of the sample sheets is performed.
+        uids = self.request.form.get("uids", [])
+        objs = map(api.get_object_by_uid, uids)
+        to_print_objs = filter(lambda obj: api.get_workflow_status_of(obj) in ["to_be_sampled", "to_be_scheduled"],
+                               objs)
+        to_print_uids = map(api.get_uid, to_print_objs)
+        if to_print_uids:
+            self.request.response.redirect(self.context.absolute_url() +
+                                           "/print_sampling_sheets?items=%s" % ",".join(to_print_uids))
```

Comment:
I don't understand that redirect here.
Why not just continue with the filtered list of UIDs, because as far as I understood it ends up with the filtered list of UIDs again here and all the filtering above is done again...